src/mito2/src/read/scan_region.rs

```diff
@@ -709,6 +709,10 @@ impl ScanInput {
         rows_in_files + rows_in_memtables
     }
 
+    pub(crate) fn predicate(&self) -> Option<Predicate> {
+        self.predicate.clone()
```

Comment:
Is predicate reference counted? Should we return a reference?

Reply:
it is